### PR TITLE
scertec: don't declare names to setec's Store

### PR DESCRIPTION
**Pull request by bradfitz**
To not block startup and defer failures to GetCertificate time.

Updates tailscale/corp#13050

### CODE REVIEW

File: scertec.go [hunk 1]
```diff
@@ -99,10 +105,20 @@ func (c *Client) GetCertificate(hello *tls.ClientHelloInfo) (*tls.Certificate, e
...
 	}
 	secName := secretName(c.prefix, hello.ServerName, typ)
+
 	sec := c.st.Secret(secName)
+
+	// In the common case the above suceeds, once setec has seen it the first time.
+	// But if it's not there, we'll look it up with a timeout.
 	if sec == nil {
-		return nil, errors.New("invalid server name")
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (creachadair) [inline scertec.go:114]: FWIW the lookups already have a timeout baked in if you don't set one (though it'll prefer yours if it's there). The baked-in one happens to be 5m already, though 🙂

File: scertec.go [hunk 2]
```diff
@@ -64,32 +66,36 @@ func NewClient(ctx context.Context, c setec.Client, cache setec.Cache, prefix st
...
-			secretName(prefix, d, "ecdsa"),
-		)
+		allowed[d] = true
 	}
 	st, err := setec.NewStore(ctx, setec.StoreConfig{
-		Client:  c,
-		Secrets: secretNames,
-		Cache:   cache,
+		Client:      c,
+		Cache:       cache,
```
Comment (creachadair) [inline scertec.go:75]: Given we already have a cache, another possible approach would be to prime the cache by hand and then use declared secrets so you don't have to keep the allowlist yourself, but this is fine too.

**Review by creachadair — APPROVED** [review]

File: scertec.go (hunk 1 above)
Comment (bradfitz) [reply scertec.go:114]: > 5m already

This is 5 _seconds_ if you misread the two fives as being the same unit.

File: scertec.go (hunk 2 above)
Comment (bradfitz) [reply scertec.go:75]: Prime the cache with what, though? I don't want to block startup, so I don't want to go reading the values from setec.

File: scertec.go (hunk 1 above)
Comment (creachadair) [reply scertec.go:114]: I did 😬

File: scertec.go (hunk 2 above)
Comment (creachadair) [reply scertec.go:75]: Even if you don't want to do a manual startup to prime the cache, you can curl the values out and bash them into JSON with jq. The format it writes to the cache is documented and I don't plan to change it in incompatible ways. (That said, I mention this mainly as another way to get the effect)

**Pull request merged**